### ISSUE TITLE
Systemtests: OpenFOAM adapter is not actually loaded

### ISSUE THREAD

**Issue by MakisH** [precice/tutorials#359]
Looking at the results of the `flow-over-heated-plate/fluid-openfoam` case, they looked strange. Then, looking at the `fluid-openfoam.log`, at the beginning of the solver execution, the issue becomes clear:

```text
--> FOAM Warning : 
    From void* Foam::dlLibraryTable::openLibrary(const Foam::fileName&, bool)
    in file db/dynamicLibrary/dlLibraryTable/dlLibraryTable.C at line 188
    Could not load "libpreciceAdapterFunctionObject.so"
libpreciceAdapterFunctionObject.so: cannot open shared object file: No such file or directory
--> FOAM Warning : 
Unknown function type preciceAdapterFunctionObject

Valid function types :
8(ObukhovLength patchProbes probes psiReactionThermoMoleFractions rhoReactionThermoMoleFractions setTimeStepFaRegion sets surfaces)


    From static Foam::autoPtr<Foam::functionObject> Foam::functionObject::New(const Foam::word&, const Foam::Time&, const Foam::dictionary&)
    in file db/functionObjects/functionObject/functionObject.C at line 129.
--> loading function object 'preCICE_Adapter'
```

This typically means that OpenFOAM cannot find the adapter library where it is supposed to be. The expected path is `$FOAM_USER_LIBBIN`, which depends on the current user and the OpenFOAM version (e.g., on my system right now, it is `~/OpenFOAM/gc-v2306/platforms/linux64GccDPInt32Opt/lib`).

@valentin-seitz Any first clue on what could be going wrong? I have not started debugging yet.

**Comment by valentin-seitz** [comment]
Its seems to be a classical user problem. 

We built the openfoam-adater with the root user, leading the .so object to be placed in: `/root/Openfoam...../lib`. 
At runtime however we enforce the use of the local user by mapping in the users. So openfoam looks in `~/openfoam.../lib` where ofc nothing can be found. 

This still bothers me. I will modify the Dockerfiles to use the precice user and enforce it via the systemtests.

**Comment by valentin-seitz** [comment]
closed by #362
